<p>hisat2-htseq-deseq2</p>
<h3 id="31-transcriptom-mapping">3.1 transcriptom mapping</h3>
<p>step 0: install tools</p>
<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>conda <span class="nb">install </span>htseq hisat2 stringtie 
</code></pre></div></div>

<p>step 1: build index and extract splice sites</p>

<p>build index</p>
<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hisat2-build <span class="nt">-p</span> <span class="o">{</span>threads<span class="o">}</span> genome/hg38.fa  hisat2_index/hg38

</code></pre></div></div>

<p>extract known splice sites for alignmnet</p>
<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hisat2_extract_splice_sites.py gencode.gtf <span class="o">&gt;</span> hisat2_index/splicesites.txt 
hisat2_extract_exons.py gencode.gtf <span class="o">&gt;</span> histat2_index/exon.txt
</code></pre></div></div>

<p>step2: mapping</p>
<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hisat2 <span class="nt">--dta</span> <span class="nt">--threads</span> <span class="k">${</span><span class="nv">threads</span><span class="k">}</span> <span class="se">\</span>
             <span class="nt">-x</span> hisat2_index/hg38 <span class="se">\</span>
             <span class="nt">--known-splicesite-infile</span> hisat2_index/splicesites.txt <span class="se">\</span>
             <span class="nt">-1</span> R1.fq.gz <span class="se">\</span>
             <span class="nt">-2</span> R2.fq.gz <span class="se">\</span>
             <span class="nt">-S</span> output.sam
</code></pre></div></div>

<p>step 3: sam to bam</p>
<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>samtools view <span class="nt">-Sbh</span>  <span class="nt">-q</span> 25 <span class="se">\</span>
              -@ <span class="k">${</span><span class="nv">threads</span><span class="k">}</span>  <span class="se">\</span>
              <span class="nt">-o</span> ouput.bam <span class="se">\</span>
              input.sam

</code></pre></div></div>

<p>step 4: bam sort and index</p>
<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>samtools <span class="nb">sort</span> -@ <span class="k">${</span><span class="nv">threads</span><span class="k">}</span> input.bam <span class="o">&gt;</span> output.sorted.bam 
samtools index input.sorted.bam <span class="c">#generate input.sorted.bam.bai</span>

</code></pre></div></div>

<p>step 5: bam to bigwig</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bamCoverage <span class="nt">-p</span> <span class="k">${</span><span class="nv">threads</span><span class="k">}</span> <span class="se">\</span>
            <span class="nt">--normalizeUsing</span> RPKM <span class="se">\ </span><span class="c"># note: other normalization options </span>
            <span class="nt">-b</span> input.sorted.bam <span class="se">\</span>
            <span class="nt">-o</span> output.bw

</code></pre></div></div>

<h3 id="32-differentially-expressed-genes-analysis">3.2 Differentially expressed genes analysis</h3>
<p>step 1: count reads</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>htseq-count <span class="nt">-r</span> pos <span class="nt">-s</span> no <span class="se">\</span>
            <span class="nt">--additional-attr</span> gene_name <span class="se">\</span>
            <span class="nt">--additional-attr</span> gene_type <span class="se">\</span>
            <span class="nt">-f</span> bam input.sorted.bam  gencode.gtf  <span class="o">&gt;</span> output.count

</code></pre></div></div>

<p>step2: differentially expressed genes analysis</p>

<p>(1) construct read count table</p>

<p>option 1: HTSeq count file input</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="s2">"DESeq2"</span><span class="p">)</span><span class="w">
</span><span class="n">directory</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"/path/to/your/readCountFiles/"</span><span class="w">
</span><span class="n">sampleFiles</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">grep</span><span class="p">(</span><span class="s2">"count"</span><span class="p">,</span><span class="w"> </span><span class="n">list.files</span><span class="p">(</span><span class="n">directory</span><span class="p">),</span><span class="w"> </span><span class="n">value</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span><span class="n">condition</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"KO"</span><span class="p">,</span><span class="s2">"KO"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="s2">"WT"</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">))</span><span class="w">
</span><span class="c1"># phenotable</span><span class="w">
</span><span class="n">sampleTable</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">sampleName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sampleFiles</span><span class="p">,</span><span class="w">
                          </span><span class="n">fileName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sampleFiles</span><span class="p">,</span><span class="w">
                          </span><span class="n">condition</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">condition</span><span class="p">)</span><span class="w">
</span><span class="c1"># construct read count table</span><span class="w">
</span><span class="n">ddsHTSeq</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DESeqDataSetFromHTSeqCount</span><span class="p">(</span><span class="n">sampleTable</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sampleTable</span><span class="p">,</span><span class="w">
                                       </span><span class="n">directory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">directory</span><span class="p">,</span><span class="w">
                                       </span><span class="n">design</span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">condition</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>option 2:  combined read count file into a single table first, then run</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span><span class="w">
</span><span class="c1"># read count table</span><span class="w">
</span><span class="n">database</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"raw.counts.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">","</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
</span><span class="n">database</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">as.matrix</span><span class="p">(</span><span class="n">database</span><span class="p">))</span><span class="w">

</span><span class="c1"># set level </span><span class="w">
</span><span class="n">condition</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"KO"</span><span class="p">,</span><span class="s2">"KO"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="s2">"WT"</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">))</span><span class="w">
</span><span class="c1"># build DESeq object</span><span class="w">
</span><span class="n">coldata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">database</span><span class="p">),</span><span class="w"> </span><span class="n">condition</span><span class="p">)</span><span class="w">
</span><span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DESeqDataSetFromMatrix</span><span class="p">(</span><span class="n">countData</span><span class="o">=</span><span class="n">database</span><span class="p">,</span><span class="w"> </span><span class="n">colData</span><span class="o">=</span><span class="n">coldata</span><span class="p">,</span><span class="w"> </span><span class="n">design</span><span class="o">=~</span><span class="n">condition</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">treatmement</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>(2) run DESeq2 and get output</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span><span class="w">
</span><span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dds</span><span class="p">[</span><span class="w"> </span><span class="n">rowSums</span><span class="p">(</span><span class="n">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">))</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w">   
</span><span class="c1"># run statistical test</span><span class="w">
</span><span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DESeq</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span><span class="w">   
</span><span class="c1"># get results  </span><span class="w">
</span><span class="n">res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">results</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span><span class="w">  
</span><span class="c1"># summary(res)  </span><span class="w">
</span><span class="n">count_r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span><span class="w"> </span><span class="n">normalized</span><span class="o">=</span><span class="nb">T</span><span class="p">)</span><span class="w">  </span><span class="c1">#normalized count matrix</span><span class="w">

</span><span class="c1"># export results</span><span class="w">
</span><span class="n">res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">res</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">res</span><span class="o">$</span><span class="n">padj</span><span class="p">),]</span><span class="w">
</span><span class="n">diff_gene</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">subset</span><span class="p">(</span><span class="n">res</span><span class="p">,</span><span class="w"> </span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="p">(</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">-1</span><span class="p">))</span><span class="w">
</span><span class="n">diff_gene</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">row.names</span><span class="p">(</span><span class="n">diff_gene</span><span class="p">)</span><span class="w">
</span><span class="n">resdata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merge</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">res</span><span class="p">),</span><span class="w"> 
                 </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span><span class="w"> </span><span class="n">normalized</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)),</span><span class="w"> 
                 </span><span class="n">by</span><span class="o">=</span><span class="s2">"row.names"</span><span class="p">,</span><span class="w">
                 </span><span class="n">sort</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span><span class="n">write.csv</span><span class="p">(</span><span class="n">resdata</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"DEGs.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

</span></code></pre></div></div>
<h4 id="33-gene-set-enrichrment-analysis">3.3 Gene set enrichrment analysis</h4>

<p>GO</p>
<ul>
  <li>clusterprofiler</li>
  <li>Enrichr (GSEApy)</li>
  <li>GSEA</li>
</ul>

<h3 id="34-alternative-splicing-analysis">3.4 Alternative splicing analysis</h3>

<p>rMATS</p>

