<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=robots content="noodp"><meta http-equiv=x-ua-compatible content="IE=edge, chrome=1"><title>Shortcut to bulk RNA-seq analysis - My Computational Genomic Playground</title><meta name=Description content="hisat2-htseq-deseq2"><meta property="og:title" content="Shortcut to bulk RNA-seq analysis"><meta property="og:description" content="hisat2-htseq-deseq2"><meta property="og:type" content="article"><meta property="og:url" content="https://zqfang.github.io/2020-01-20-rnaseq/"><meta property="og:image" content="https://zqfang.github.io/logo.png"><meta property="article:published_time" content="2020-01-20T00:00:00+00:00"><meta property="article:modified_time" content="2020-01-20T00:00:00+00:00"><meta name=twitter:card content="summary_large_image"><meta name=twitter:image content="https://zqfang.github.io/logo.png"><meta name=twitter:title content="Shortcut to bulk RNA-seq analysis"><meta name=twitter:description content="hisat2-htseq-deseq2"><meta name=application-name content="#root max >"><meta name=apple-mobile-web-app-title content="#root max >"><meta name=theme-color content="#ffffff"><meta name=msapplication-TileColor content="#da532c"><link rel="shortcut icon" type=image/x-icon href=/favicon.ico><link rel=icon type=image/png sizes=32x32 href=/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=/favicon-16x16.png><link rel=apple-touch-icon sizes=180x180 href=/apple-touch-icon.png><link rel=mask-icon href=/safari-pinned-tab.svg color=#5bbad5><link rel=manifest href=/site.webmanifest><link rel=canonical href=https://zqfang.github.io/2020-01-20-rnaseq/><link rel=prev href=https://zqfang.github.io/2020-01-20-chip/><link rel=next href=https://zqfang.github.io/2020-01-20-begin/><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/normalize.css@8.0.1/normalize.min.css><link rel=stylesheet href=/css/style.min.1e2694bed152fa2922dbe909a441838ed693d88b1330f97485bfa8ed78da42df.css integrity="sha256-HiaUvtFS+iki2+kJpEGDjtaT2IsTMPl0hb+o7XjaQt8="><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.13.0/css/all.min.css><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/animate.css@3.7.2/animate.min.css><script type=application/ld+json>{"@context":"http://schema.org","@type":"BlogPosting","headline":"Shortcut to bulk RNA-seq analysis","inLanguage":"en","mainEntityOfPage":{"@type":"WebPage","@id":"https:\/\/zqfang.github.io\/2020-01-20-rnaseq\/"},"image":["https:\/\/zqfang.github.io\/images\/Apple-Devices-Preview.png"],"genre":"posts","keywords":"RNA-seq","wordcount":431,"url":"https:\/\/zqfang.github.io\/2020-01-20-rnaseq\/","datePublished":"2020-01-20T00:00:00+00:00","dateModified":"2020-01-20T00:00:00+00:00","license":"This work is licensed under a Creative Commons Attribution-NonCommercial 4.0 International License.","publisher":{"@type":"Organization","name":"xxxx","logo":"https:\/\/zqfang.github.io\/images\/avatar.png"},"author":{"@type":"Person","name":"zqfang"},"description":"hisat2-htseq-deseq2"}</script></head><body header-desktop=fixed header-mobile=auto><script type=text/javascript>(window.localStorage&&localStorage.getItem('theme')?localStorage.getItem('theme')==='dark':('auto'==='auto'?window.matchMedia('(prefers-color-scheme: dark)').matches:'auto'==='dark'))&&document.body.setAttribute('theme','dark');</script><div id=mask></div><div class=wrapper><header class=desktop id=header-desktop><div class=header-wrapper><div class=header-title><a href=/ title="My Computational Genomic Playground"><span class=header-title-pre><i class="fas fa-hashtag"></i></span>MAX <span class=header-title-post><i class="fas fa-terminal"></i></span></a></div><div class=menu><div class=menu-inner><a class=menu-item href=/posts/>Posts </a><a class=menu-item href=/categories/>Categories </a><a class=menu-item href=/publication/>Publication </a><a class=menu-item href=/portfolio/>Portfolio </a><a class=menu-item href=/tags/>Tags </a><a class=menu-item href=https://github.com/zqfang title=GitHub rel="noopener noreffer" target=_blank><i class="fab fa-github fa-fw"></i></a><a class=menu-item href=/about/ title=About>About </a><span class="menu-item delimiter"></span><span class="menu-item search" id=search-desktop><input type=text placeholder="Search titles or contents..." id=search-input-desktop>
<a href=javascript:void(0); class="search-button search-toggle" id=search-toggle-desktop title=Search><i class="fas fa-search fa-fw"></i></a><a href=javascript:void(0); class="search-button search-clear" id=search-clear-desktop title=Clear><i class="fas fa-times-circle fa-fw"></i></a><span class="search-button search-loading" id=search-loading-desktop><i class="fas fa-spinner fa-fw fa-spin"></i></span></span><a href=javascript:void(0); class="menu-item theme-switch" title="Switch Theme"><i class="fas fa-adjust fa-fw"></i></a></div></div></div></header><header class=mobile id=header-mobile><div class=header-container><div class=header-wrapper><div class=header-title><a href=/ title="My Computational Genomic Playground"><span class=header-title-pre><i class="fas fa-hashtag"></i></span>MAX <span class=header-title-post><i class="fas fa-terminal"></i></span></a></div><div class=menu-toggle id=menu-toggle-mobile><span></span><span></span><span></span></div></div><div class=menu id=menu-mobile><div class=search-wrapper><div class="search mobile" id=search-mobile><input type=text placeholder="Search titles or contents..." id=search-input-mobile>
<a href=javascript:void(0); class="search-button search-toggle" id=search-toggle-mobile title=Search><i class="fas fa-search fa-fw"></i></a><a href=javascript:void(0); class="search-button search-clear" id=search-clear-mobile title=Clear><i class="fas fa-times-circle fa-fw"></i></a><span class="search-button search-loading" id=search-loading-mobile><i class="fas fa-spinner fa-fw fa-spin"></i></span></div><a href=javascript:void(0); class=search-cancel id=search-cancel-mobile>Cancel</a></div><a class=menu-item href=/posts/>Posts</a><a class=menu-item href=/categories/>Categories</a><a class=menu-item href=/publication/>Publication</a><a class=menu-item href=/portfolio/>Portfolio</a><a class=menu-item href=/tags/>Tags</a><a class=menu-item href=https://github.com/zqfang title=GitHub rel="noopener noreffer" target=_blank><i class="fab fa-github fa-fw"></i></a><a class=menu-item href=/about/ title=About>About</a><a href=javascript:void(0); class="menu-item theme-switch" title="Switch Theme">
<i class="fas fa-adjust fa-fw"></i></a></div></div></header><div class="search-dropdown desktop"><div id=search-dropdown-desktop></div></div><div class="search-dropdown mobile"><div id=search-dropdown-mobile></div></div><main class=main><div class=container><div class=toc id=toc-auto><h2 class=toc-title>Contents</h2><div class=toc-content id=toc-content-auto></div></div><article class="page single"><h1 class="single-title animated flipInX">Shortcut to bulk RNA-seq analysis</h1><div class=post-meta><div class=post-meta-line><span class=post-author><a href=/ title=Author rel=author class=author><i class="fas fa-user-circle fa-fw"></i>zqfang</a></span>&nbsp;<span class=post-category>included in <a href=/categories/make-bioinfo-uncool-again/><i class="far fa-folder fa-fw"></i>Make bioinfo uncool again</a></span></div><div class=post-meta-line><i class="far fa-calendar-alt fa-fw"></i>&nbsp;<time datetime=2020-01-20>2020-01-20</time>&nbsp;<i class="fas fa-pencil-alt fa-fw"></i>&nbsp;431 words&nbsp;
<i class="far fa-clock fa-fw"></i>&nbsp;3 minutes&nbsp;</div></div><div class="details toc" id=toc-static kept><div class="details-summary toc-title"><span>Contents</span>
<span><i class="details-icon fas fa-angle-right"></i></span></div><div class="details-content toc-content" id=toc-content-static><nav id=TableOfContents><ul><li><ul><li><a href=#31-transcriptom-mapping>3.1 transcriptom mapping</a></li><li><a href=#32-differentially-expressed-genes-analysis>3.2 Differentially expressed genes analysis</a><ul><li><a href=#33-gene-set-enrichrment-analysis>3.3 Gene set enrichrment analysis</a></li></ul></li><li><a href=#34-alternative-splicing-analysis>3.4 Alternative splicing analysis</a></li></ul></li></ul></nav></div></div><div class=content id=content><p>hisat2-htseq-deseq2</p><h3 id=31-transcriptom-mapping>3.1 transcriptom mapping</h3><p>step 0: install tools</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>conda install htseq hisat2 stringtie 
</code></pre></td></tr></table></div></div><p>step 1: build index and extract splice sites</p><p>build index</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>hisat2-build -p <span class=o>{</span>threads<span class=o>}</span> genome/hg38.fa  hisat2_index/hg38

</code></pre></td></tr></table></div></div><p>extract known splice sites for alignmnet</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>hisat2_extract_splice_sites.py gencode.gtf &gt; hisat2_index/splicesites.txt 
hisat2_extract_exons.py gencode.gtf &gt; histat2_index/exon.txt
</code></pre></td></tr></table></div></div><p>step2: mapping</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span><span class=lnt>6
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>hisat2 --dta --threads <span class=si>${</span><span class=nv>threads</span><span class=si>}</span> <span class=se>\
</span><span class=se></span>             -x hisat2_index/hg38 <span class=se>\
</span><span class=se></span>             --known-splicesite-infile hisat2_index/splicesites.txt <span class=se>\
</span><span class=se></span>             -1 R1.fq.gz <span class=se>\
</span><span class=se></span>             -2 R2.fq.gz <span class=se>\
</span><span class=se></span>             -S output.sam
</code></pre></td></tr></table></div></div><p>step 3: sam to bam</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>samtools view -Sbh  -q <span class=m>25</span> <span class=se>\
</span><span class=se></span>              -@ <span class=si>${</span><span class=nv>threads</span><span class=si>}</span>  <span class=se>\
</span><span class=se></span>              -o ouput.bam <span class=se>\
</span><span class=se></span>              input.sam

</code></pre></td></tr></table></div></div><p>step 4: bam sort and index</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>samtools sort -@ <span class=si>${</span><span class=nv>threads</span><span class=si>}</span> input.bam &gt; output.sorted.bam 
samtools index input.sorted.bam <span class=c1>#generate input.sorted.bam.bai</span>

</code></pre></td></tr></table></div></div><p>step 5: bam to bigwig</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>bamCoverage -p <span class=si>${</span><span class=nv>threads</span><span class=si>}</span> <span class=se>\
</span><span class=se></span>            --normalizeUsing RPKM <span class=se>\ </span><span class=c1># note: other normalization options</span> 
            -b input.sorted.bam <span class=se>\
</span><span class=se></span>            -o output.bw

</code></pre></td></tr></table></div></div><h3 id=32-differentially-expressed-genes-analysis>3.2 Differentially expressed genes analysis</h3><p>step 1: count reads</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-shell data-lang=shell>htseq-count -r pos -s no <span class=se>\
</span><span class=se></span>            --additional-attr gene_name <span class=se>\
</span><span class=se></span>            --additional-attr gene_type <span class=se>\
</span><span class=se></span>            -f bam input.sorted.bam  gencode.gtf  &gt; output.count

</code></pre></td></tr></table></div></div><p>step2: differentially expressed genes analysis</p><p>(1) construct read count table</p><p>option 1: HTSeq count file input</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span><span class=lnt>11
</span><span class=lnt>12
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=nf>library</span><span class=p>(</span><span class=s>&#34;DESeq2&#34;</span><span class=p>)</span>
<span class=n>directory</span> <span class=o>&lt;-</span> <span class=s>&#34;/path/to/your/readCountFiles/&#34;</span>
<span class=n>sampleFiles</span> <span class=o>&lt;-</span> <span class=nf>grep</span><span class=p>(</span><span class=s>&#34;count&#34;</span><span class=p>,</span> <span class=nf>list.files</span><span class=p>(</span><span class=n>directory</span><span class=p>),</span> <span class=n>value</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>)</span>
<span class=n>condition</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=nf>c</span><span class=p>(</span><span class=s>&#34;KO&#34;</span><span class=p>,</span><span class=s>&#34;KO&#34;</span><span class=p>,</span> <span class=s>&#34;WT&#34;</span><span class=p>,</span><span class=s>&#34;WT&#34;</span><span class=p>),</span> <span class=n>levels</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;WT&#34;</span><span class=p>,</span> <span class=s>&#34;KO&#34;</span><span class=p>))</span>
<span class=c1># phenotable</span>
<span class=n>sampleTable</span> <span class=o>&lt;-</span> <span class=nf>data.frame</span><span class=p>(</span><span class=n>sampleName</span> <span class=o>=</span> <span class=n>sampleFiles</span><span class=p>,</span>
                          <span class=n>fileName</span> <span class=o>=</span> <span class=n>sampleFiles</span><span class=p>,</span>
                          <span class=n>condition</span> <span class=o>=</span> <span class=n>condition</span><span class=p>)</span>
<span class=c1># construct read count table</span>
<span class=n>ddsHTSeq</span> <span class=o>&lt;-</span> <span class=nf>DESeqDataSetFromHTSeqCount</span><span class=p>(</span><span class=n>sampleTable</span> <span class=o>=</span> <span class=n>sampleTable</span><span class=p>,</span>
                                       <span class=n>directory</span> <span class=o>=</span> <span class=n>directory</span><span class=p>,</span>
                                       <span class=n>design</span><span class=o>=</span> <span class=o>~</span> <span class=n>condition</span><span class=p>)</span>
</code></pre></td></tr></table></div></div><p>option 2: combined read count file into a single table first, then run</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=nf>library</span><span class=p>(</span><span class=n>DESeq2</span><span class=p>)</span>
<span class=c1># read count table</span>
<span class=n>database</span> <span class=o>&lt;-</span> <span class=nf>read.table</span><span class=p>(</span><span class=n>file</span> <span class=o>=</span> <span class=s>&#34;raw.counts.csv&#34;</span><span class=p>,</span> <span class=n>sep</span> <span class=o>=</span> <span class=s>&#34;,&#34;</span><span class=p>,</span> <span class=n>header</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=n>row.names</span> <span class=o>=</span> <span class=m>1</span><span class=p>)</span>
<span class=n>database</span> <span class=o>&lt;-</span> <span class=nf>round</span><span class=p>(</span><span class=nf>as.matrix</span><span class=p>(</span><span class=n>database</span><span class=p>))</span>

<span class=c1># set level </span>
<span class=n>condition</span> <span class=o>&lt;-</span> <span class=nf>factor</span><span class=p>(</span><span class=nf>c</span><span class=p>(</span><span class=s>&#34;KO&#34;</span><span class=p>,</span><span class=s>&#34;KO&#34;</span><span class=p>,</span> <span class=s>&#34;WT&#34;</span><span class=p>,</span><span class=s>&#34;WT&#34;</span><span class=p>),</span> <span class=n>levels</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;WT&#34;</span><span class=p>,</span> <span class=s>&#34;KO&#34;</span><span class=p>))</span>
<span class=c1># build DESeq object</span>
<span class=n>coldata</span> <span class=o>&lt;-</span> <span class=nf>data.frame</span><span class=p>(</span><span class=n>row.names</span> <span class=o>=</span> <span class=nf>colnames</span><span class=p>(</span><span class=n>database</span><span class=p>),</span> <span class=n>condition</span><span class=p>)</span>
<span class=n>dds</span> <span class=o>&lt;-</span> <span class=nf>DESeqDataSetFromMatrix</span><span class=p>(</span><span class=n>countData</span><span class=o>=</span><span class=n>database</span><span class=p>,</span> <span class=n>colData</span><span class=o>=</span><span class=n>coldata</span><span class=p>,</span> <span class=n>design</span><span class=o>=~</span><span class=n>condition</span> <span class=o>+</span> <span class=n>treatmement</span><span class=p>)</span>
</code></pre></td></tr></table></div></div><p>(2) run DESeq2 and get output</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span><span class=lnt>11
</span><span class=lnt>12
</span><span class=lnt>13
</span><span class=lnt>14
</span><span class=lnt>15
</span><span class=lnt>16
</span><span class=lnt>17
</span><span class=lnt>18
</span><span class=lnt>19
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=nf>library</span><span class=p>(</span><span class=n>DESeq2</span><span class=p>)</span>
<span class=n>dds</span> <span class=o>&lt;-</span> <span class=n>dds[</span> <span class=nf>rowSums</span><span class=p>(</span><span class=nf>counts</span><span class=p>(</span><span class=n>dds</span><span class=p>))</span> <span class=o>&gt;</span> <span class=m>1</span><span class=p>,</span> <span class=n>]</span>   
<span class=c1># run statistical test</span>
<span class=n>dds</span> <span class=o>&lt;-</span> <span class=nf>DESeq</span><span class=p>(</span><span class=n>dds</span><span class=p>)</span>   
<span class=c1># get results  </span>
<span class=n>res</span> <span class=o>&lt;-</span> <span class=nf>results</span><span class=p>(</span><span class=n>dds</span><span class=p>)</span>  
<span class=c1># summary(res)  </span>
<span class=n>count_r</span> <span class=o>&lt;-</span> <span class=nf>counts</span><span class=p>(</span><span class=n>dds</span><span class=p>,</span> <span class=n>normalized</span><span class=o>=</span><span class=bp>T</span><span class=p>)</span>  <span class=c1>#normalized count matrix</span>

<span class=c1># export results</span>
<span class=n>res</span> <span class=o>&lt;-</span> <span class=n>res</span><span class=nf>[order</span><span class=p>(</span><span class=n>res</span><span class=o>$</span><span class=n>padj</span><span class=p>),</span><span class=n>]</span>
<span class=n>diff_gene</span> <span class=o>&lt;-</span> <span class=nf>subset</span><span class=p>(</span><span class=n>res</span><span class=p>,</span> <span class=n>padj</span> <span class=o>&lt;</span> <span class=m>0.05</span> <span class=o>&amp;</span> <span class=p>(</span><span class=n>log2FoldChange</span> <span class=o>&gt;</span> <span class=m>1</span> <span class=o>|</span> <span class=n>log2FoldChange</span> <span class=o>&lt;</span> <span class=m>-1</span><span class=p>))</span>
<span class=n>diff_gene</span> <span class=o>&lt;-</span> <span class=nf>row.names</span><span class=p>(</span><span class=n>diff_gene</span><span class=p>)</span>
<span class=n>resdata</span> <span class=o>&lt;-</span> <span class=nf>merge</span><span class=p>(</span><span class=nf>as.data.frame</span><span class=p>(</span><span class=n>res</span><span class=p>),</span> 
                 <span class=nf>as.data.frame</span><span class=p>(</span><span class=nf>counts</span><span class=p>(</span><span class=n>dds</span><span class=p>,</span> <span class=n>normalized</span><span class=o>=</span><span class=kc>TRUE</span><span class=p>)),</span> 
                 <span class=n>by</span><span class=o>=</span><span class=s>&#34;row.names&#34;</span><span class=p>,</span>
                 <span class=n>sort</span><span class=o>=</span><span class=kc>FALSE</span><span class=p>)</span>
<span class=nf>write.csv</span><span class=p>(</span><span class=n>resdata</span><span class=p>,</span> <span class=n>file</span> <span class=o>=</span> <span class=s>&#34;DEGs.csv&#34;</span><span class=p>,</span> <span class=n>row.names</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>

</code></pre></td></tr></table></div></div><h4 id=33-gene-set-enrichrment-analysis>3.3 Gene set enrichrment analysis</h4><p>GO</p><ul><li>clusterprofiler</li><li>Enrichr (GSEApy)</li><li>GSEA</li></ul><h3 id=34-alternative-splicing-analysis>3.4 Alternative splicing analysis</h3><p>rMATS</p></div><div class=post-footer id=post-footer><div class=post-info><div class=post-info-line><div class=post-info-mod><span>Updated on 2020-01-20</span></div><div class=post-info-license></div></div><div class=post-info-line><div class=post-info-md></div><div class=post-info-share><span><a href=javascript:void(0); title="Share on Twitter" data-sharer=twitter data-url=https://zqfang.github.io/2020-01-20-rnaseq/ data-title="Shortcut to bulk RNA-seq analysis" data-hashtags=RNA-seq><i class="fab fa-twitter fa-fw"></i></a><a href=javascript:void(0); title="Share on Facebook" data-sharer=facebook data-url=https://zqfang.github.io/2020-01-20-rnaseq/ data-hashtag=RNA-seq><i class="fab fa-facebook-square fa-fw"></i></a><a href=javascript:void(0); title="Share on Hacker News" data-sharer=hackernews data-url=https://zqfang.github.io/2020-01-20-rnaseq/ data-title="Shortcut to bulk RNA-seq analysis"><i class="fab fa-hacker-news fa-fw"></i></a><a href=javascript:void(0); title="Share on Line" data-sharer=line data-url=https://zqfang.github.io/2020-01-20-rnaseq/ data-title="Shortcut to bulk RNA-seq analysis"><i data-svg-src=https://cdn.jsdelivr.net/npm/simple-icons@2.14.0/icons/line.svg></i></a><a href=javascript:void(0); title="Share on 微博" data-sharer=weibo data-url=https://zqfang.github.io/2020-01-20-rnaseq/ data-title="Shortcut to bulk RNA-seq analysis"><i class="fab fa-weibo fa-fw"></i></a></span></div></div></div><div class=post-info-more><section class=post-tags><i class="fas fa-tags fa-fw"></i>&nbsp;<a href=/tags/rna-seq/>RNA-seq</a></section><section><span><a href=javascript:void(0); onclick=window.history.back();>Back</a></span>&nbsp;|&nbsp;<span><a href=/>Home</a></span></section></div><div class=post-nav><a href=/2020-01-20-chip/ class=prev rel=prev title="Shortcut to ChIP-seq analysis"><i class="fas fa-angle-left fa-fw"></i>Shortcut to ChIP-seq analysis</a>
<a href=/2020-01-20-begin/ class=next rel=next title="Cheatsheet for command line">Cheatsheet for command line<i class="fas fa-angle-right fa-fw"></i></a></div></div><div id=comments><div id=disqus_thread class=comment></div><noscript>Please enable JavaScript to view the comments powered by <a href=https://disqus.com/?ref_noscript>Disqus</a>.</noscript></div></article></div></main><footer class=footer><div class=footer-container><div class=footer-line><i class="far fa-copyright fa-fw"></i><span itemprop=copyrightYear>2020</span><span class=author itemprop=copyrightHolder>&nbsp;<a href=/ target=_blank>zqfang</a></span>&nbsp;|&nbsp;<span class=license><a rel="license external nofollow noopener noreffer" href=https://creativecommons.org/licenses/by-nc/4.0/ target=_blank>CC BY-NC 4.0</a></span></div></div></footer></div><div id=fixed-buttons><a href=# id=back-to-top class=fixed-button title="Back to Top"><i class="fas fa-arrow-up fa-fw"></i></a><a href=# id=view-comments class=fixed-button title="View Comments"><i class="fas fa-comment fa-fw"></i></a></div><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/copy-tex.min.css><script type=text/javascript src=https://bioninja-1.disqus.com/embed.js defer></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/smooth-scroll@16.1.3/dist/smooth-scroll.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/autocomplete.js@0.37.1/dist/autocomplete.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/lunr@2.3.8/lunr.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/lazysizes@5.2.2/lazysizes.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/sharer.js@0.4.0/sharer.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/copy-tex.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/mhchem.min.js></script><script type=text/javascript>window.config={"code":{"copyTitle":"Copy to clipboard","maxShownLines":50},"comment":{},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"highlightTag":"em","lunrIndexURL":"/index.json","maxResultLength":10,"noResultsFound":"No results found","snippetLength":30,"type":"lunr"}};</script><script type=text/javascript src=/js/theme.min.f51938f3065a40ee841bcb558e4330e31fd26c0ea55343fff8770b88b0319a3c.js integrity="sha256-9Rk48wZaQO6EG8tVjkMw4x/SbA6lU0P/+HcLiLAxmjw="></script></body></html>