<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=robots content="noodp"><meta http-equiv=x-ua-compatible content="IE=edge, chrome=1"><title>Convert Seurat to Scanpy h5ad - My Computational Genomic Playground</title><meta name=Description content="The correct way to convert seurat to Scanpy h5ad"><meta property="og:title" content="Convert Seurat to Scanpy h5ad"><meta property="og:description" content="The correct way to convert seurat to Scanpy h5ad"><meta property="og:type" content="article"><meta property="og:url" content="https://zqfang.github.io/2020-04-28-seurat2scanpy/"><meta property="og:image" content="https://zqfang.github.io/logo.png"><meta property="article:published_time" content="2023-02-21T00:00:00+00:00"><meta property="article:modified_time" content="2023-02-21T00:00:00+00:00"><meta name=twitter:card content="summary_large_image"><meta name=twitter:image content="https://zqfang.github.io/logo.png"><meta name=twitter:title content="Convert Seurat to Scanpy h5ad"><meta name=twitter:description content="The correct way to convert seurat to Scanpy h5ad"><meta name=application-name content="Pleiades"><meta name=apple-mobile-web-app-title content="Pleiades"><meta name=theme-color content="#ffffff"><meta name=msapplication-TileColor content="#da532c"><link rel="shortcut icon" type=image/x-icon href=/favicon.ico><link rel=icon type=image/png sizes=32x32 href=/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=/favicon-16x16.png><link rel=apple-touch-icon sizes=180x180 href=/apple-touch-icon.png><link rel=mask-icon href=/safari-pinned-tab.svg color=#5bbad5><link rel=manifest href=/site.webmanifest><link rel=canonical href=https://zqfang.github.io/2020-04-28-seurat2scanpy/><link rel=prev href=https://zqfang.github.io/2022-08-10-bda1/><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/normalize.css@8.0.1/normalize.min.css><link rel=stylesheet href=/css/style.min.1e2694bed152fa2922dbe909a441838ed693d88b1330f97485bfa8ed78da42df.css integrity="sha256-HiaUvtFS+iki2+kJpEGDjtaT2IsTMPl0hb+o7XjaQt8="><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.13.0/css/all.min.css><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/animate.css@3.7.2/animate.min.css><script type=application/ld+json>{"@context":"http://schema.org","@type":"BlogPosting","headline":"Convert Seurat to Scanpy h5ad","inLanguage":"en","mainEntityOfPage":{"@type":"WebPage","@id":"https:\/\/zqfang.github.io\/2020-04-28-seurat2scanpy\/"},"image":["https:\/\/zqfang.github.io\/images\/Apple-Devices-Preview.png"],"genre":"posts","keywords":"scRNA-seq, Scanpy, Seurat, Bioinformatics","wordcount":912,"url":"https:\/\/zqfang.github.io\/2020-04-28-seurat2scanpy\/","datePublished":"2023-02-21T00:00:00+00:00","dateModified":"2023-02-21T00:00:00+00:00","license":"This work is licensed under a Creative Commons Attribution-NonCommercial 4.0 International License.","publisher":{"@type":"Organization","name":"xxxx","logo":"https:\/\/zqfang.github.io\/images\/avatar.png"},"author":{"@type":"Person","name":"zqfang"},"description":"The correct way to convert seurat to Scanpy h5ad"}</script></head><body header-desktop=fixed header-mobile=auto><script type=text/javascript>(window.localStorage&&localStorage.getItem('theme')?localStorage.getItem('theme')==='dark':('auto'==='auto'?window.matchMedia('(prefers-color-scheme: dark)').matches:'auto'==='dark'))&&document.body.setAttribute('theme','dark');</script><div id=mask></div><div class=wrapper><header class=desktop id=header-desktop><div class=header-wrapper><div class=header-title><a href=/ title="My Computational Genomic Playground"><span class=header-title-pre><span>&#8711;</span></span>Pleiades <span class=header-title-post><i class="fas fa-terminal"></i></span></a></div><div class=menu><div class=menu-inner><a class=menu-item href=/posts/>Posts </a><a class=menu-item href=/categories/>Categories </a><a class=menu-item href=/publication/>Publication </a><a class=menu-item href=/portfolio/>Portfolio </a><a class=menu-item href=/tags/>Tags </a><a class=menu-item href=/about/ title=About>About </a><a class=menu-item href=https://github.com/zqfang title=GitHub rel="noopener noreffer" target=_blank><i class="fab fa-github fa-fw"></i></a><span class="menu-item delimiter"></span><span class="menu-item search" id=search-desktop><input type=text placeholder="Search titles or contents..." id=search-input-desktop>
<a href=javascript:void(0); class="search-button search-toggle" id=search-toggle-desktop title=Search><i class="fas fa-search fa-fw"></i></a><a href=javascript:void(0); class="search-button search-clear" id=search-clear-desktop title=Clear><i class="fas fa-times-circle fa-fw"></i></a><span class="search-button search-loading" id=search-loading-desktop><i class="fas fa-spinner fa-fw fa-spin"></i></span></span><a href=javascript:void(0); class="menu-item theme-switch" title="Switch Theme"><i class="fas fa-adjust fa-fw"></i></a></div></div></div></header><header class=mobile id=header-mobile><div class=header-container><div class=header-wrapper><div class=header-title><a href=/ title="My Computational Genomic Playground"><span class=header-title-pre><span>&#8711;</span></span>Pleiades <span class=header-title-post><i class="fas fa-terminal"></i></span></a></div><div class=menu-toggle id=menu-toggle-mobile><span></span><span></span><span></span></div></div><div class=menu id=menu-mobile><div class=search-wrapper><div class="search mobile" id=search-mobile><input type=text placeholder="Search titles or contents..." id=search-input-mobile>
<a href=javascript:void(0); class="search-button search-toggle" id=search-toggle-mobile title=Search><i class="fas fa-search fa-fw"></i></a><a href=javascript:void(0); class="search-button search-clear" id=search-clear-mobile title=Clear><i class="fas fa-times-circle fa-fw"></i></a><span class="search-button search-loading" id=search-loading-mobile><i class="fas fa-spinner fa-fw fa-spin"></i></span></div><a href=javascript:void(0); class=search-cancel id=search-cancel-mobile>Cancel</a></div><a class=menu-item href=/posts/>Posts</a><a class=menu-item href=/categories/>Categories</a><a class=menu-item href=/publication/>Publication</a><a class=menu-item href=/portfolio/>Portfolio</a><a class=menu-item href=/tags/>Tags</a><a class=menu-item href=/about/ title=About>About</a><a class=menu-item href=https://github.com/zqfang title=GitHub rel="noopener noreffer" target=_blank><i class="fab fa-github fa-fw"></i></a><a href=javascript:void(0); class="menu-item theme-switch" title="Switch Theme">
<i class="fas fa-adjust fa-fw"></i></a></div></div></header><div class="search-dropdown desktop"><div id=search-dropdown-desktop></div></div><div class="search-dropdown mobile"><div id=search-dropdown-mobile></div></div><main class=main><div class=container><div class=toc id=toc-auto><h2 class=toc-title>Contents</h2><div class=toc-content id=toc-content-auto></div></div><article class="page single"><h1 class="single-title animated flipInX">Convert Seurat to Scanpy h5ad</h1><div class=post-meta><div class=post-meta-line><span class=post-author><a href=/ title=Author rel=author class=author><i class="fas fa-user-circle fa-fw"></i>zqfang</a></span>&nbsp;<span class=post-category>included in <a href=/categories/make-bioinfo-uncool-again/><i class="far fa-folder fa-fw"></i>Make bioinfo uncool again</a></span></div><div class=post-meta-line><i class="far fa-calendar-alt fa-fw"></i>&nbsp;<time datetime=2023-02-21>2023-02-21</time>&nbsp;<i class="fas fa-pencil-alt fa-fw"></i>&nbsp;912 words&nbsp;
<i class="far fa-clock fa-fw"></i>&nbsp;5 minutes&nbsp;</div></div><div class="details toc" id=toc-static kept><div class="details-summary toc-title"><span>Contents</span>
<span><i class="details-icon fas fa-angle-right"></i></span></div><div class="details-content toc-content" id=toc-content-static><nav id=TableOfContents><ul><li><a href=#mudataseurat>MuDataSeurat</a><ul><li><a href=#install>Install</a></li><li><a href=#usage>Usage</a><ul><li><a href=#optional-step-1-slim-down-a-seurat-object>(Optional) Step 1: Slim down a Seurat object.</a></li><li><a href=#step-2-write-seurat-to-h5ad>Step 2. Write seurat to h5ad</a></li></ul></li></ul></li><li><a href=#seuratdisk>SeuratDisk</a></li><li><a href=#seurat---loom---scanpy>Seurat -> loom -> scanpy</a></li><li><a href=#rpy2>rpy2</a><ul><li><a href=#1-install-seurat-v302-or-python-kernel-will-always-died>1. Install <code>Seurat v3.0.2</code>, or python kernel will always died!!!</a></li><li><a href=#2-set-the-r-version-for-rpy2>2. Set the R version for <code>rpy2</code></a></li><li><a href=#3-now-youre-good-to-go>3. Now, you&rsquo;re good to go</a></li></ul></li></ul></nav></div></div><div class=content id=content><p>IMPORTANT UPDDATE: 2023-02-21</p><h2 id=mudataseurat>MuDataSeurat</h2><p><strong>Recommended!!!</strong></p><p>MuDataSeurat write h5ad file directly. easy install and simple use</p><p>Refer to <a href=https://pmbio.github.io/MuDataSeurat/index.html target=_blank rel="noopener noreffer">MuDataSeurat</a></p><h3 id=install>Install</h3><p>I have add some extra features. Please <strong>Install my fork</strong> which works for anndata >=0.8</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=c1># in R console</span>
<span class=c1>## install myfork, which compatible with latest anndata </span>
<span class=n>remotes</span><span class=o>::</span><span class=nf>install_github</span><span class=p>(</span><span class=s>&#34;zqfang/MuDataSeurat&#34;</span><span class=p>,</span> <span class=n>force</span> <span class=o>=</span> <span class=bp>T</span><span class=p>)</span>
</code></pre></td></tr></table></div></div><h3 id=usage>Usage</h3><h4 id=optional-step-1-slim-down-a-seurat-object>(Optional) Step 1: Slim down a Seurat object.</h4><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span><span class=lnt>11
</span><span class=lnt>12
</span><span class=lnt>13
</span><span class=lnt>14
</span><span class=lnt>15
</span><span class=lnt>16
</span><span class=lnt>17
</span><span class=lnt>18
</span><span class=lnt>19
</span><span class=lnt>20
</span><span class=lnt>21
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=c1># (optional) step 1: Slim down a Seurat object. So you get raw counts, lognorm counts</span>
<span class=n>seu</span> <span class=o>=</span> <span class=nf>DietSeurat</span><span class=p>(</span>
  <span class=n>srt</span><span class=p>,</span>
  <span class=n>counts</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=c1># so, raw counts save to adata.layers[&#39;counts&#39;]</span>
  <span class=n>data</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=c1># so, log1p counts save to adata.X when scale.data = False, else adata.layers[&#39;data&#39;]</span>
  <span class=n>scale.data</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>,</span> <span class=c1># set to false</span>
  <span class=n>features</span> <span class=o>=</span> <span class=nf>rownames</span><span class=p>(</span><span class=n>srt</span><span class=p>),</span> <span class=c1># export all genes, not just top highly variable genes</span>
  <span class=n>assays</span> <span class=o>=</span> <span class=s>&#34;RNA&#34;</span><span class=p>,</span>
  <span class=n>dimreducs</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;pca&#34;</span><span class=p>,</span><span class=s>&#34;umap&#34;</span><span class=p>),</span>
  <span class=n>graphs</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;RNA_nn&#34;</span><span class=p>,</span> <span class=s>&#34;RNA_snn&#34;</span><span class=p>),</span> <span class=c1># to RNA_nn -&gt; distances, RNA_snn -&gt; connectivities</span>
  <span class=n>misc</span> <span class=o>=</span> <span class=kc>TRUE</span>
<span class=p>)</span>


<span class=c1># .X, .layers[&#39;counts&#39;]. .raw.X</span>
<span class=c1># Assumptions:</span>
<span class=c1>#   1. counts/data/scale.data -&gt; X</span>
<span class=c1>#   3. counts &amp; data -&gt; layers[&#39;counts&#39;], X</span>
<span class=c1>#   2. data &amp; scale.data -&gt; layers[&#39;data&#39;], X</span>
<span class=c1>#   4. counts &amp; scale.data -&gt; layers[&#39;counts&#39;], X</span>
<span class=c1>#   5. counts &amp; data &amp; scale.data -&gt; layers[&#39;counts&#39;], layers[&#39;data&#39;], X</span>
</code></pre></td></tr></table></div></div><h4 id=step-2-write-seurat-to-h5ad>Step 2. Write seurat to h5ad</h4><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span><span class=lnt>11
</span><span class=lnt>12
</span><span class=lnt>13
</span><span class=lnt>14
</span><span class=lnt>15
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=c1># single modality</span>
<span class=n>MuDataSeurat</span><span class=o>::</span><span class=nf>WriteH5AD</span><span class=p>(</span><span class=n>srt</span><span class=p>,</span> <span class=s>&#34;srt.h5ad&#34;</span><span class=p>,</span> <span class=n>assay</span><span class=o>=</span><span class=s>&#34;RNA&#34;</span><span class=p>)</span>
<span class=c1># multi modality, ATAC+RNA, CITE-seq </span>
<span class=n>MuDataSeurat</span><span class=o>::</span><span class=nf>WriteH5MU</span><span class=p>(</span><span class=n>srt</span><span class=p>,</span> <span class=s>&#34;srt.h5mu&#34;</span><span class=p>)</span>

<span class=c1>## data mapping</span>
<span class=c1>## GEX</span>
<span class=c1># counts -&gt; adata.layers[&#39;counts&#39;]</span>
<span class=c1># data -&gt; adata.layers[&#39;data&#39;] # log1p data</span>
<span class=c1># scale.data -&gt; adata.X</span>
<span class=c1># adata.raw.X set to None</span>
<span class=c1>## Graph</span>
<span class=c1># RNA_nn -&gt; adata.obsp[&#39;nn&#39;] # this is knn adjmat. the old .uns[&#39;neighbors&#39;][&#39;distance&#39;]</span>
<span class=c1># RNA_snn -&gt; adata.obsp[&#39;snn] # umap graph. the old .uns[&#39;neighbors&#39;][&#39;connectivies&#39;]</span>
<span class=c1>## </span>
</code></pre></td></tr></table></div></div><ul><li><code>ReadH5AD()</code>: Read an .h5mu file and create a Seurat object.</li><li><code>ReadH5MU()</code>: Create a Seurat object from .h5mu file contents</li><li><code>WriteH5AD()</code>: Write one assay to .h5ad</li><li><code>WriteH5MU()</code>: Create an .h5mu file with data from a Seurat object</li></ul><h2 id=seuratdisk>SeuratDisk</h2><p>Please see <a href=https://mojaveazure.github.io/seurat-disk/reference/Convert.html target=_blank rel="noopener noreffer">SeuratDisk</a> to convert seurat to scanpy.</p><p>Not recommend, since it&rsquo;s not fully compatitble with anndata standards.</p><p><strong>Tips</strong>:</p><ol><li>set default assay to <code>RNA</code> before covert to h5ad.</li><li>if raw read count need to be imported to anndata, you should only contain counts slot in your seurat object before convertion</li></ol><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span><span class=lnt>11
</span><span class=lnt>12
</span><span class=lnt>13
</span><span class=lnt>14
</span><span class=lnt>15
</span><span class=lnt>16
</span><span class=lnt>17
</span><span class=lnt>18
</span><span class=lnt>19
</span><span class=lnt>20
</span><span class=lnt>21
</span><span class=lnt>22
</span><span class=lnt>23
</span><span class=lnt>24
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=nf>library</span><span class=p>(</span><span class=n>Seurat</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>SeuratDisk</span><span class=p>)</span>

<span class=c1># step 1: Slim down a Seurat object. So you get raw counts, lognorm counts</span>

<span class=n>seu</span> <span class=o>=</span> <span class=nf>DietSeurat</span><span class=p>(</span>
  <span class=n>srt</span><span class=p>,</span>
  <span class=n>counts</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=c1># so, raw counts save to adata.raw.X </span>
  <span class=n>data</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=c1># so, log1p counts save to adata.X</span>
  <span class=n>scale.data</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>,</span> <span class=c1># set to false, or else will save to adata.X</span>
  <span class=n>features</span> <span class=o>=</span> <span class=nf>rownames</span><span class=p>(</span><span class=n>srt</span><span class=p>),</span> <span class=c1># export all genes, not just top highly variable genes</span>
  <span class=n>assays</span> <span class=o>=</span> <span class=s>&#34;RNA&#34;</span><span class=p>,</span>
  <span class=n>dimreducs</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;pca&#34;</span><span class=p>,</span><span class=s>&#34;umap&#34;</span><span class=p>),</span>
  <span class=n>graphs</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&#34;RNA_nn&#34;</span><span class=p>,</span> <span class=s>&#34;RNA_snn&#34;</span><span class=p>),</span> <span class=c1># to RNA_nn -&gt; distances, RNA_snn -&gt; connectivities</span>
  <span class=n>misc</span> <span class=o>=</span> <span class=kc>TRUE</span>
<span class=p>)</span>

<span class=c1># step 2: factor to character, or else your factor will be number in adata </span>
<span class=n>i</span> <span class=o>&lt;-</span> <span class=nf>sapply</span><span class=p>(</span><span class=n>seu</span><span class=o>@</span><span class=n>meta.data</span><span class=p>,</span> <span class=n>is.factor</span><span class=p>)</span>
<span class=n>seu</span><span class=o>@</span><span class=n>meta.data[i]</span> <span class=o>&lt;-</span> <span class=nf>lapply</span><span class=p>(</span><span class=n>seu</span><span class=o>@</span><span class=n>meta.data[i]</span><span class=p>,</span> <span class=n>as.character</span><span class=p>)</span>

<span class=c1># step 3: convert </span>
<span class=nf>SaveH5Seurat</span><span class=p>(</span><span class=n>seu</span><span class=p>,</span> <span class=n>filename</span> <span class=o>=</span> <span class=s>&#34;srt.h5seurat&#34;</span><span class=p>,</span> <span class=n>overwrite</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>)</span>
<span class=nf>Convert</span><span class=p>(</span><span class=s>&#34;srt.h5seurat&#34;</span><span class=p>,</span> <span class=s>&#34;srt.h5ad&#34;</span><span class=p>,</span> <span class=n>assay</span><span class=o>=</span><span class=s>&#34;RNA&#34;</span><span class=p>,</span> <span class=n>overwrite</span> <span class=o>=</span> <span class=kc>TRUE</span><span class=p>)</span>
</code></pre></td></tr></table></div></div><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span><span class=lnt>6
</span><span class=lnt>7
</span><span class=lnt>8
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-python data-lang=python><span class=c1># load h5ad</span>
<span class=kn>import</span> <span class=nn>scanpy</span> <span class=kn>as</span> <span class=nn>sc</span>
<span class=n>adata</span> <span class=o>=</span> <span class=n>sc</span><span class=o>.</span><span class=n>read_h5ad</span><span class=p>(</span><span class=s2>&#34;srt.h5ad&#34;</span><span class=p>)</span>
<span class=c1># save counts to layers</span>
<span class=n>adata</span><span class=o>.</span><span class=n>layers</span><span class=p>[</span><span class=s1>&#39;counts&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>adata</span><span class=o>.</span><span class=n>raw</span><span class=o>.</span><span class=n>X</span><span class=o>.</span><span class=n>copy</span><span class=p>()</span>
<span class=n>adata</span><span class=o>.</span><span class=n>layers</span><span class=p>[</span><span class=s1>&#39;log1p&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>data</span><span class=o>.</span><span class=n>X</span><span class=o>.</span><span class=n>copy</span><span class=p>()</span>

<span class=c1># you need to scale data for downsream tasks if needed.</span>
</code></pre></td></tr></table></div></div><h2 id=seurat---loom---scanpy>Seurat -> loom -> scanpy</h2><p><strong>NOT RECOMMENED</strong></p><p>You actually neeed additional steps when seurat -> loom -> scanpy</p><p>see <a href=https://github.com/basilkhuder/Seurat-to-RNA-Velocity target=_blank rel="noopener noreffer">here</a></p><ol><li>save to <code>loom</code> format.</li></ol><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span><span class=lnt>6
</span><span class=lnt>7
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=n>pbmc.loom</span> <span class=o>&lt;-</span> <span class=nf>as.loom</span><span class=p>(</span><span class=n>seurat_object</span><span class=p>,</span> <span class=n>filename</span> <span class=o>=</span> <span class=s>&#34;../output/pbmc.loom&#34;</span><span class=p>,</span> <span class=n>verbose</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>
<span class=n>pbmc.loom</span><span class=o>$</span><span class=nf>close_all</span><span class=p>()</span> <span class=c1># alway close when done </span>

<span class=nf>write.csv</span><span class=p>(</span><span class=nf>Cells</span><span class=p>(</span><span class=n>seurat_object</span><span class=p>),</span> <span class=n>file</span> <span class=o>=</span> <span class=s>&#34;cellID_obs.csv&#34;</span><span class=p>,</span> <span class=n>row.names</span> <span class=o>=</span> <span class=kc>FALSE</span><span class=p>)</span>
<span class=nf>write.csv</span><span class=p>(</span><span class=nf>Embeddings</span><span class=p>(</span><span class=n>seurat_object</span><span class=p>,</span> <span class=n>reduction</span> <span class=o>=</span> <span class=s>&#34;umap&#34;</span><span class=p>),</span> <span class=n>file</span> <span class=o>=</span> <span class=s>&#34;cell_embeddings.csv&#34;</span><span class=p>)</span>
<span class=nf>write.csv</span><span class=p>(</span><span class=n>seurat_object</span><span class=o>@</span><span class=n>meta.data</span><span class=o>$</span><span class=n>seurat_clusters</span><span class=p>,</span> <span class=n>file</span> <span class=o>=</span> <span class=s>&#34;clusters.csv&#34;</span><span class=p>)</span>

</code></pre></td></tr></table></div></div><ol start=2><li>read into scanpy</li></ol><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span><span class=lnt>11
</span><span class=lnt>12
</span><span class=lnt>13
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-python data-lang=python><span class=kn>import</span> <span class=nn>scanpy</span> <span class=kn>as</span> <span class=nn>sc</span>
<span class=n>adata</span> <span class=o>=</span> <span class=n>sc</span><span class=o>.</span><span class=n>read_loom</span><span class=p>(</span><span class=s2>&#34;../output/pbmc.loom&#34;</span><span class=p>)</span>

<span class=n>sample_obs</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=s2>&#34;cellID_obs.csv&#34;</span><span class=p>)</span>
<span class=n>umap_cord</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=s2>&#34;cell_embeddings.csv&#34;</span><span class=p>)</span>
<span class=n>cell_clusters</span> <span class=o>=</span> <span class=n>pd</span><span class=o>.</span><span class=n>read_csv</span><span class=p>(</span><span class=s2>&#34;clusters_obs.csv&#34;</span><span class=p>)</span>


<span class=c1># now add metadata to the adata </span>
<span class=c1># e.g.</span>
<span class=n>adata</span><span class=o>.</span><span class=n>obsm</span><span class=p>[</span><span class=s1>&#39;X_umap&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=n>umap_ordered</span><span class=o>.</span><span class=n>values</span> <span class=c1># cellID should be matched first</span>
<span class=n>adata</span><span class=o>.</span><span class=n>uns</span><span class=p>[</span><span class=s1>&#39;Cluster_colors&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=o>...</span>
<span class=o>...</span>
</code></pre></td></tr></table></div></div><ol start=3><li>open loom in R</li></ol><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-R data-lang=R><span class=n>immune</span> <span class=o>&lt;-</span> <span class=nf>Connect</span><span class=p>(</span><span class=n>filename</span> <span class=o>=</span> <span class=s>&#34;../data/mmune_cells.loom&#34;</span><span class=p>,</span> <span class=n>mode</span> <span class=o>=</span> <span class=s>&#34;r&#34;</span><span class=p>)</span>
<span class=n>seurat</span> <span class=o>&lt;-</span> <span class=nf>as.Seurat</span><span class=p>(</span><span class=n>immune</span><span class=p>)</span>
<span class=n>immune</span><span class=o>$</span><span class=nf>close_all</span><span class=p>()</span> <span class=c1># alway close when done </span>
</code></pre></td></tr></table></div></div><h2 id=rpy2>rpy2</h2><p>This is the old way. Very hard to make it work. <strong>Not recommended!</strong></p><p>Convert <code>Seurat</code> to <code>Scanpy</code> costed me a lot of time to convert seurat objects to scanpy. It&rsquo;s not a pleasant experience.</p><h3 id=1-install-seurat-v302-or-python-kernel-will-always-died>1. Install <code>Seurat v3.0.2</code>, or python kernel will always died!!!</h3><p>Don&rsquo;t know why latest seurat not work.</p><h3 id=2-set-the-r-version-for-rpy2>2. Set the R version for <code>rpy2</code></h3><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span><span class=lnt>6
</span><span class=lnt>7
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-python data-lang=python><span class=c1># user defined R installation</span>
<span class=kn>import</span> <span class=nn>os</span>
<span class=c1># path to your libR.so, only Seurat v3.0.2 works! </span>
<span class=c1># create a conda R env for seurat 3.0.2 first</span>
<span class=n>os</span><span class=o>.</span><span class=n>environ</span><span class=p>[</span><span class=s1>&#39;R_HOME&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=s1>&#39;/home/fangzq/miniconda/envs/seurat/lib/R&#39;</span> 
<span class=c1># path depends on where you installed Python.</span>
<span class=n>os</span><span class=o>.</span><span class=n>environ</span><span class=p>[</span><span class=s1>&#39;R_USER&#39;</span><span class=p>]</span> <span class=o>=</span> <span class=s1>&#39;/home/fangzq/miniconda/lib/python3.7/site-packages/rpy2&#39;</span> 
</code></pre></td></tr></table></div></div><h3 id=3-now-youre-good-to-go>3. Now, you&rsquo;re good to go</h3><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-python data-lang=python><span class=kn>import</span> <span class=nn>scanpy</span> <span class=kn>as</span> <span class=nn>sc</span>
<span class=kn>import</span> <span class=nn>glob</span>
</code></pre></td></tr></table></div></div><p>Install <code>anndata2ri</code> first</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span><span class=lnt>3
</span><span class=lnt>4
</span><span class=lnt>5
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-python data-lang=python><span class=kn>import</span> <span class=nn>anndata2ri</span>
<span class=kn>from</span> <span class=nn>rpy2.robjects</span> <span class=kn>import</span> <span class=n>r</span>
<span class=kn>from</span> <span class=nn>rpy2.robjects.conversion</span> <span class=kn>import</span> <span class=n>localconverter</span>
<span class=c1># activate rpy2 env</span>
<span class=n>anndata2ri</span><span class=o>.</span><span class=n>activate</span><span class=p>()</span>
</code></pre></td></tr></table></div></div><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-python data-lang=python><span class=n>robjs</span> <span class=o>=</span> <span class=n>glob</span><span class=o>.</span><span class=n>glob</span><span class=p>(</span><span class=s2>&#34;data/*Robj&#34;</span><span class=p>)</span>
</code></pre></td></tr></table></div></div><p>Convert to h5ad</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-python data-lang=python><span class=n>r</span><span class=p>(</span><span class=s1>&#39;library(Seurat)&#39;</span><span class=p>)</span>
<span class=k>for</span> <span class=n>robj</span> <span class=ow>in</span> <span class=n>robjs</span><span class=p>:</span>
    <span class=n>r</span><span class=p>(</span><span class=n>f</span><span class=s1>&#39;x&lt;-load(&#34;{robj}&#34;)&#39;</span><span class=p>)</span>
    <span class=n>r</span><span class=p>(</span><span class=s1>&#39;y=get(x)&#39;</span><span class=p>)</span>
    <span class=n>r</span><span class=p>(</span><span class=s1>&#39;rm(x)&#39;</span><span class=p>)</span>
    <span class=n>r</span><span class=p>(</span><span class=s1>&#39;DefaultAssay(y) &lt;- &#34;RNA&#34;&#39;</span><span class=p>)</span> <span class=c1># get raw count matrix to save</span>
    <span class=c1># seurat2 object</span>
    <span class=c1># adata = r(&#39;as.SingleCellExperiment(UpdateSeuratObject(y))&#39;)</span>
    <span class=n>adata</span> <span class=o>=</span> <span class=n>r</span><span class=p>(</span><span class=s1>&#39;as.SingleCellExperiment(y)&#39;</span><span class=p>)</span>
    <span class=n>adata</span><span class=o>.</span><span class=n>write_h5ad</span><span class=p>(</span><span class=n>filename</span><span class=o>=</span><span class=n>robj</span><span class=o>.</span><span class=n>replace</span><span class=p>(</span><span class=s2>&#34;Robj&#34;</span><span class=p>,</span><span class=s2>&#34;h5ad&#34;</span><span class=p>))</span>
</code></pre></td></tr></table></div></div></div><div class=post-footer id=post-footer><div class=post-info><div class=post-info-line><div class=post-info-mod><span>Updated on 2023-02-21</span></div><div class=post-info-license></div></div><div class=post-info-line><div class=post-info-md></div><div class=post-info-share><span><a href=javascript:void(0); title="Share on Twitter" data-sharer=twitter data-url=https://zqfang.github.io/2020-04-28-seurat2scanpy/ data-title="Convert Seurat to Scanpy h5ad" data-hashtags=scRNA-seq,Scanpy,Seurat,Bioinformatics><i class="fab fa-twitter fa-fw"></i></a><a href=javascript:void(0); title="Share on Facebook" data-sharer=facebook data-url=https://zqfang.github.io/2020-04-28-seurat2scanpy/ data-hashtag=scRNA-seq><i class="fab fa-facebook-square fa-fw"></i></a><a href=javascript:void(0); title="Share on Hacker News" data-sharer=hackernews data-url=https://zqfang.github.io/2020-04-28-seurat2scanpy/ data-title="Convert Seurat to Scanpy h5ad"><i class="fab fa-hacker-news fa-fw"></i></a><a href=javascript:void(0); title="Share on Line" data-sharer=line data-url=https://zqfang.github.io/2020-04-28-seurat2scanpy/ data-title="Convert Seurat to Scanpy h5ad"><i data-svg-src=https://cdn.jsdelivr.net/npm/simple-icons@2.14.0/icons/line.svg></i></a><a href=javascript:void(0); title="Share on 微博" data-sharer=weibo data-url=https://zqfang.github.io/2020-04-28-seurat2scanpy/ data-title="Convert Seurat to Scanpy h5ad"><i class="fab fa-weibo fa-fw"></i></a></span></div></div></div><div class=post-info-more><section class=post-tags><i class="fas fa-tags fa-fw"></i>&nbsp;<a href=/tags/scrna-seq/>scRNA-seq</a>,&nbsp;<a href=/tags/scanpy/>Scanpy</a>,&nbsp;<a href=/tags/seurat/>Seurat</a>,&nbsp;<a href=/tags/bioinformatics/>Bioinformatics</a></section><section><span><a href=javascript:void(0); onclick=window.history.back();>Back</a></span>&nbsp;|&nbsp;<span><a href=/>Home</a></span></section></div><div class=post-nav><a href=/2022-08-10-bda1/ class=prev rel=prev title="Bayesian Data Analysis: Basics"><i class="fas fa-angle-left fa-fw"></i>Bayesian Data Analysis: Basics</a></div></div><div id=comments><div id=disqus_thread class=comment></div><noscript>Please enable JavaScript to view the comments powered by <a href=https://disqus.com/?ref_noscript>Disqus</a>.</noscript></div></article></div></main><footer class=footer><div class=footer-container><div class=footer-line><i class="far fa-copyright fa-fw"></i><span itemprop=copyrightYear>2020 - 2024</span><span class=author itemprop=copyrightHolder>&nbsp;<a href=/ target=_blank>zqfang</a></span>&nbsp;|&nbsp;<span class=license><a rel="license external nofollow noopener noreffer" href=https://creativecommons.org/licenses/by-nc/4.0/ target=_blank>CC BY-NC 4.0</a></span></div></div></footer></div><div id=fixed-buttons><a href=# id=back-to-top class=fixed-button title="Back to Top"><i class="fas fa-arrow-up fa-fw"></i></a><a href=# id=view-comments class=fixed-button title="View Comments"><i class="fas fa-comment fa-fw"></i></a></div><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/copy-tex.min.css><script type=text/javascript src=https://bioninja-1.disqus.com/embed.js defer></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/smooth-scroll@16.1.3/dist/smooth-scroll.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/autocomplete.js@0.37.1/dist/autocomplete.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/lunr@2.3.8/lunr.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/lazysizes@5.2.2/lazysizes.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/sharer.js@0.4.0/sharer.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/copy-tex.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/mhchem.min.js></script><script type=text/javascript>window.config={"code":{"copyTitle":"Copy to clipboard","maxShownLines":50},"comment":{},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"highlightTag":"em","lunrIndexURL":"/index.json","maxResultLength":10,"noResultsFound":"No results found","snippetLength":30,"type":"lunr"}};</script><script type=text/javascript src=/js/theme.min.f51938f3065a40ee841bcb558e4330e31fd26c0ea55343fff8770b88b0319a3c.js integrity="sha256-9Rk48wZaQO6EG8tVjkMw4x/SbA6lU0P/+HcLiLAxmjw="></script></body></html>